treeherder/perf/management/commands/remove_vcs_data.py

```diff
@@ -0,0 +1,43 @@
+"""
+TODO: Remove this module entirely once all vcs data
+has been deleted from databases across all Treeherder
+environments (from prototype up to & including production).
+"""
+import time
+from datetime import timedelta
+
+from django.core.management.base import BaseCommand
+
+from treeherder.model.data_cycling import MaxRuntime
+from treeherder.perf.exceptions import MaxRuntimeExceeded
+from treeherder.perf.models import PerformanceSignature
+
+
+class Command(BaseCommand):
+    help = "Remove all vcs data ingested by Perfherder"
+
+    # max runtime
+    PER_DELETE_SPRINT = timedelta(minutes=5)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__timer = MaxRuntime(self.PER_DELETE_SPRINT)
+        self.__timer.start_timer()
+
+    def handle(self, *args, **options):
+        vcs_signatures = PerformanceSignature.objects.filter(framework__name='vcs')
+        for signature in vcs_signatures:
+            signature.delete()  # intentionally cascades to data points also
```

Comment:
I'm guessing this will be a long running task executed via the scheduler. Would it make sense to use the `chunked_queryset` and bulk delete instead, and then you can log the status with each chunk deleted?

Reply:
Not actually. We need to manually run this command once, on all our Heroku environments (production included). We want to purge all vcs data we've collected so far (we won't collect anymore of it, as this PR also disables its ingestion).

After that, we can remove the `remove_vcs_data.py` module.

Thus, this PR doesn't need to be integrated into our data expiring algorithm.

Reply:
Right, I understand that. But how will you know when it's complete? Are you planning to manually query the db for these frameworks a day or two or three after executing the command in the scheduler? If it was interrupted and didn't complete deleting everything, how will you know what caused the interruption? :)

Reply:
Good question. I prepared [this SQL query](https://sql.telemetry.mozilla.org/queries/78218/source) for checking how much data is left. This will let us know if `remove_vcs_data` finished deleting all `vcs` data.

If the command gets interrupted, we'll see the trace back into our console. It should have enough context.
Still, I think in 85% of the cases the solution would be to run `./manage.py remove_vcs_data` again, to continue deleting where it left off.

Reply:
So I've talked with @davehunt about this. Turns out we don't really need to manually remove the `vcs` data we've collected thus far.
Our data expiring algorithms will automatically take care of that.